File: src/types/annotation.ts
import type { ExtractedMetadata } from './paper';

export interface PdfRect {
  x: number;
  y: number;
  width: number;
  height: number;
}

export interface Annotation {
  id: string;
  paperId: string;
  field: keyof ExtractedMetadata | 'title' | 'abstract' | 'notes';
  text: string;
  pageNumber: number;
  rects: PdfRect[];
  createdAt: string;
}

File: src-tauri/src/models/annotation.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PdfRect {
    pub x: f64,
    pub y: f64,
    pub width: f64,
    pub height: f64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Annotation {
    pub id: String,
    pub paper_id: String,
    pub field: String,
    pub text: String,
    pub page_number: i32,
    pub rects: Vec<PdfRect>,
    pub created_at: String,
}
